src/store/utils.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * @format
+ * @flow
+ */
+
+// Gutenberg imports
+import { parse, serialize } from '@wordpress/blocks';
+
+import md5 from 'md5';
+import { Store } from 'redux';
+
+import type { BlockType, StateType } from './';
+
+export function html2State( html: string ) {
+	const blocksFromHtml = parse( html );
+	const reserialized = blocks2html( blocksFromHtml );
+	const hash = md5( reserialized );
+	const state: StateType = {
```

Comment:
just an idea: we can make use of ES6 classes for State to make use of constructors, getter&setters, for example every time the html is set we can update blocks and initialHtmlHash inside the class. or we can define a constructor that is taking html as input and update blocks and initialHtmlHash in that constructor. again this is just an idea current version is also ok for me.

Reply:
Let me see if I understand the proposal: Will that class update its internal vars?

If yes, I'm not sure that's what we want for Redux. My understanding is that the reducer is supposed to return a new "state" object and avoid mutating the old one, isn't it? In that case, I'm not sure a State class would be too useful since we would essentially only use its constructor, and that's pretty much what `html2State` performs.

Let me know if there is something in the proposal that I'm missing, thanks!

Reply:
Oh, and btw, we might don't want to spend too much time iterating on this since we will be using GB's stores soon #226 anyway.

Reply:
You are right about mutating. we could figure it out though maybe there's an easy way to clone a class object. I'll investigate this further on a more free time and If I discover a useful practice I will post a p2 about it.